cpp/include/cudf/functions.h

```diff
@@ -382,64 +382,20 @@ gdf_error gdf_hash_partition(int num_input_cols,
+ * @Param[in] inp Input column for prefix sum with null_count = 0
+ * @Param[out] out The output column containing the prefix sum of the input
+ * @Param inclusive[in] Flag for applying an inclusive prefix sum if non-zero,
+ * an exclusive prefix sum if 0.
  * 
- * @Returns   GDF_SUCCESS if the operation was successful, otherwise an appropriate
- * error code. If inp->null_count is not set to 0 GDF_VALIDITY_UNSUPPORTED is
- * returned.
- */
-/* ----------------------------------------------------------------------------*/
-gdf_error gdf_prefixsum_generic(gdf_column *inp, gdf_column *out, int inclusive);
-
-/* --------------------------------------------------------------------------*/
-/** 
- * @Synopsis  Computes the prefix sum of a column
- * 
- * @Param inp Input column for prefix sum with null_count = 0
- * @Param out The output column containing the prefix sum of the input
- * @Param inclusive Flag for applying an inclusive prefix sum
- * 
- * @Returns   GDF_SUCCESS if the operation was successful, otherwise an appropriate
- * error code. If inp->null_count is not set to 0 GDF_VALIDITY_UNSUPPORTED is
- * returned.
- */
-/* ----------------------------------------------------------------------------*/
-gdf_error gdf_prefixsum_i8(gdf_column *inp, gdf_column *out, int inclusive);
-
-/* --------------------------------------------------------------------------*/
-/** 
- * @Synopsis  Computes the prefix sum of a column
- * 
- * @Param inp Input column for prefix sum with null_count = 0
- * @Param out The output column containing the prefix sum of the input
- * @Param inclusive Flag for applying an inclusive prefix sum
- * 
- * @Returns   GDF_SUCCESS if the operation was successful, otherwise an appropriate
- * error code. If inp->null_count is not set to 0 GDF_VALIDITY_UNSUPPORTED is
- * returned.
+ * @Returns   GDF_SUCCESS if the operation was successful,
+ * otherwise an appropriate error code. 
+ * GDF_VALIDITY_UNSUPPORTED if inp->null_count is not set to 0.
+ * GDF_UNSUPPORTED_DTYPE if inp->dtype is not an arithmetic type.
  */
 /* ----------------------------------------------------------------------------*/
-gdf_error gdf_prefixsum_i32(gdf_column *inp, gdf_column *out, int inclusive);
-
-/* --------------------------------------------------------------------------*/
-/** 
- * @Synopsis  Computes the prefix sum of a column
- * 
- * @Param inp Input column for prefix sum with null_count = 0
- * @Param out The output column containing the prefix sum of the input
- * @Param inclusive Flag for applying an inclusive prefix sum
- * 
- * @Returns   GDF_SUCCESS if the operation was successful, otherwise an appropriate
- * error code. If inp->null_count is not set to 0 GDF_VALIDITY_UNSUPPORTED is
- * returned.
- */
-/* ----------------------------------------------------------------------------*/
-gdf_error gdf_prefixsum_i64(gdf_column *inp, gdf_column *out, int inclusive);
-
+gdf_error gdf_prefixsum(gdf_column *inp, gdf_column *out, int inclusive);
```

Comment:
But why is inclusive an `int` rather than a `bool`? C99 has `bool`s, C++ has `bool`s, why not use it?

Reply:
I keep it because cudf/functions.h looks like C fashioned header file.
But I just found gdf_digitize() alreay has bool parameter.
I will use bool in next commit.
Thanks!